wandb/integration/catboost/catboost.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+"""
+catboost init
+"""
+
+import wandb
+
+
+class WandbCallback:
+    """`WandbCallback` automatically integrates CatBoost with wandb.
+
+    Arguments:
+        - metric_period: (int) if you are passing `metric_period` to your CatBoost model please pass the same value here (default=1).
+
+    Passing `WandbCallback` to CatBoost will:
+    - log training and validation metrics at every `metric_period`
+    - log iteration at every `metric_period`
+
+    Example:
+        ```
+        train_pool = Pool(train[features], label=train['label'], cat_features=cat_features)
+        test_pool = Pool(test[features], label=test['label'], cat_features=cat_features)
+
+        model = CatBoostRegressor(iterations=100,
+            loss_function='Cox',
+            eval_metric='Cox',
+        )
+
+        model.fit(train_pool,
+                  eval_set=test_pool,
+                  callbacks=[WandbCallback()])
+        ```
+    """
+
+    def __init__(self, metric_period: int = 1):
+        if wandb.run is None:
+            raise wandb.Error("You must call wandb.init() before WandbCallback()")
```

Comment:
Could you please add a test case for this? Add a try/except clause to your functional test? We're trying to please codecov as much as possible...